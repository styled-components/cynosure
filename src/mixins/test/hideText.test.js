// @flow
import hideText from '../hideText'

describe('hideText', () => {
  it('should return the CSS in JS', () => {
    expect({ ...hideText() }).toEqual({
      overflow: 'hidden',
      textIndent: '101%',
      whiteSpace: 'nowrap',
    })
  })

  it('should add rules when block has existing rules', () => {
    expect({
      ...hideText(),
      'background-image': 'url(logo.png)',
    }).toEqual({
      'background-image': 'url(logo.png)',
      overflow: 'hidden',
      textIndent: '101%',
      whiteSpace: 'nowrap',
    })
  })
})
