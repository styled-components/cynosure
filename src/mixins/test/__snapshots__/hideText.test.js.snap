exports[`hideText should add rules when block has existing rules 1`] = `
Object {
  "background-image": "url(logo.png)",
  "overflow": "hidden",
  "textIndent": "101%",
  "whiteSpace": "nowrap",
}
`;

exports[`hideText should return the CSS in JS 1`] = `
Object {
  "overflow": "hidden",
  "textIndent": "101%",
  "whiteSpace": "nowrap",
}
`;
